service/handlers_keys.go

```diff
@@ -0,0 +1,131 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016-2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package service
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+)
+
+// AuthorizedKeysResponse is the JSON response from the API AuthorizedKeys call.
+// This returns a slice containing the list of ssh keys that can access the vault.
+type AuthorizedKeysResponse struct {
+	Keys []string `json:"keys"`
+}
+
+// BooleanResponse is the JSON response from an API method, indicating success or failure.
+type BooleanResponse struct {
+	Success      bool   `json:"success"`
+	ErrorMessage string `json:"message"`
+}
+
+// AuthorizedKey is the JSON body with the public key.
+type AuthorizedKey struct {
+	PublicKey string `json:"device-key"`
```

Comment:
The JSON tag should be ssh-key or public-key. device-key is confusing with other term